src/persist-client/src/internal/machine.rs

```diff
@@ -721,6 +721,9 @@ where
                 return x;
             }
             Err(err) => {
+                if err.is_timeout() {
+                    metrics.timeouts.inc();
```

Comment:
is it useful to capture this data here? we don't record the other reasons for retrying, though timeouts can be good to know about.

I also tried making the trait ext require a metric, like `fut.timeout(duration, counter)` but the more I went down that road the more it felt like duplicating the work we already do to capture metrics per consensus/blob op.